ipa-core/src/cli/hpke.rs

```diff
@@ -0,0 +1,337 @@
+    secret_sharing::IntoShares,
+    test_fixture::{ipa::TestRawDataRecord, Reconstruct},
+};
+
+type BreakdownKey = BA8;
+type Timestamp = BA20;
+type TriggerValue = BA3;
+
+#[derive(Debug, Parser)]
+#[clap(name = "test_encrypt", about = "Test Encrypt")]
+#[command(about)]
+pub struct EncryptArgs {
+    /// Path to file to secret share and encrypt
+    #[arg(long)]
+    input_file: PathBuf,
+    // /// The destination dir for encrypted output.
+    // /// In that dir, it will create helper1.enc,
+    // /// helper2.enc, and helper3.enc
+    #[arg(long, value_name = "FILE")]
+    output_dir: PathBuf,
+    /// Path to helper network configuration file
+    #[arg(long)]
+    network: PathBuf,
+}
+
+#[derive(Debug, Parser)]
+#[clap(name = "test_decrypt", about = "Test Decrypt")]
+#[command(about)]
+pub struct DecryptArgs {
+    /// Path to helper1 file to decrypt
+    #[arg(long)]
+    input_file1: PathBuf,
+
+    /// Helper1 Private key for decrypting match keys
+    #[arg(long)]
+    mk_private_key1: PathBuf,
+
+    /// Path to helper2 file to decrypt
+    #[arg(long)]
+    input_file2: PathBuf,
+
+    /// Helper2 Private key for decrypting match keys
+    #[arg(long)]
+    mk_private_key2: PathBuf,
+
+    /// Path to helper3 file to decrypt
+    #[arg(long)]
+    input_file3: PathBuf,
+
+    /// Helper3 Private key for decrypting match keys
+    #[arg(long)]
+    mk_private_key3: PathBuf,
+
+    /// The destination file for decrypted output.
+    #[arg(long, value_name = "FILE")]
+    output_file: PathBuf,
+}
+
+/// # Panics
+/// if input file or network file are not correctly formatted
+/// # Errors
+/// if it cannot open the files
+pub fn encrypt(args: &EncryptArgs) -> Result<(), BoxError> {
+    let input = InputSource::from_file(&args.input_file);
+    let records = input.iter::<TestRawDataRecord>().collect::<Vec<_>>();
+
+    let mut rng = StdRng::from_entropy();
+    let mut key_registries = KeyRegistries::default();
+    let network = NetworkConfig::from_toml_str(&read_to_string(&args.network).unwrap()).unwrap();
+    let Some((key_id, key_registries)) = key_registries.init_from(&network, DEFAULT_KEY_ID) else {
+        panic!("could not load network file")
+    };
+    let shares: [Vec<OprfReport<BreakdownKey, TriggerValue, Timestamp>>; 3] =
+        records.iter().cloned().share();
+
+    let shares_key_registries = zip(shares, key_registries);
+
+    for (index, (shares, key_registry)) in shares_key_registries.enumerate() {
+        let mut writer: Box<dyn Write> = Box::new(
+            OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(args.output_dir.join(format!("helper{}.enc", index + 1)))?,
+        );
+        for share in shares {
+            let output = share.encrypt(key_id, key_registry, &mut rng).unwrap();
+            let hex_output = hex::encode(&output);
+            writeln!(writer, "{hex_output}")?;
+        }
+    }
```

Comment:
```suggestion
 for (index, (shares, key_registry)) in zip(shares, key_registries).enumerate() {
        let mut writer =
            OpenOptions::new()
                .write(true)
                .create_new(true)
                .open(args.output_dir.join(format!("helper{}.enc", index + 1)))?;

        for share in shares {
            let output = share.encrypt(key_id, key_registry, &mut rng)?;
            let hex_output = hex::encode(&output);
            writeln!(writer, "{hex_output}")?;
        }
    }
```

Reply:
also consider handling file open error - I believe the default error does not mention file name, so it is also very unhelpful to debug